server/core/src/main/java/org/infinispan/server/core/admin/AdminServerTask.java

```diff
@@ -0,0 +1,66 @@
+package org.infinispan.server.core.admin;
+
+import java.lang.invoke.MethodHandles;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.server.core.logging.Log;
+import org.infinispan.tasks.ServerTask;
+import org.infinispan.tasks.TaskContext;
+import org.infinispan.util.logging.LogFactory;
+
+import io.netty.util.CharsetUtil;
+
+/**
+ * Common base for admin server tasks
+ *
+ * @author Tristan Tarrant
+ * @since 9.0
+ */
+
+public abstract class AdminServerTask<T> implements ServerTask<T> {
+   protected EmbeddedCacheManager cacheManager;
+   protected final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass(), Log.class);
+   private Map<String, String> parameters;
+   private EnumSet<AdminFlag> flags;
+
+   @Override
+   public final String getName() {
+      return "@@" + getTaskContextName() + "@" + getTaskOperationName();
+   }
+
+   @Override
+   public final void setTaskContext(TaskContext taskContext) {
+      this.cacheManager = taskContext.getCacheManager();
+      Map<String, byte[]> rawParams = (Map<String, byte[]>) taskContext.getParameters().get();
+      parameters = new HashMap<>(rawParams.size());
+      rawParams.entrySet().stream().forEach(entry -> {
```

Comment:
You can use collect here as well. http://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-